### ISSUE TITLE
remove_attachment breaks on Rails 3.2

### ISSUE THREAD

**Issue by iwz** [thoughtbot/paperclip#1987]
I'm using Rails 3.2.21, paperclip 4.3.0, mysql 14.14

Given this migration:

```ruby
class AddAttachmentAssetToLessonContents < ActiveRecord::Migration
  def change
    add_attachment :lesson_contents, :asset
  end
end
```

After `rake db:migrate` runs successfully, `rake db:rollback` fails with the following:

```
Mysql2::Error: Can't DROP 'string'; check that column/key exists: ALTER TABLE `lesson_contents` DROP `string`/Users/iwz/.rvm/gems/ruby-2.1.6/gems/rack-mini-profiler-0.1.27/Ruby/lib/patches/sql_patches.rb:45:in `query'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/rack-mini-profiler-0.1.27/Ruby/lib/patches/sql_patches.rb:45:in `query'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `block in execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in `block in log'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in `log'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/mysql2_adapter.rb:213:in `execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:279:in `block in remove_column'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:278:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:278:in `remove_column'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:44:in `block (2 levels) in remove_attachment'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:42:in `each_pair'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:42:in `block in remove_attachment'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:40:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:40:in `remove_attachment'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:466:in `block in method_missing'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:438:in `block in say_with_time'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:438:in `say_with_time'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:458:in `method_missing'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:402:in `block (4 levels) in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:401:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:401:in `block (3 levels) in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:358:in `revert'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:400:in `block (2 levels) in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:399:in `block in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:129:in `with_connection'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:389:in `migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:528:in `migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:720:in `block (2 levels) in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:777:in `call'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:777:in `ddl_transaction'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:719:in `block in migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:700:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:700:in `migrate'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:574:in `down'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:656:in `move'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/migration.rb:562:in `rollback'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/railties/databases.rake:276:in `block (2 levels) in <top (required)>'
/Users/iwz/.rvm/gems/ruby-2.1.6/bin/ruby_executable_hooks:15:in `eval'
/Users/iwz/.rvm/gems/ruby-2.1.6/bin/ruby_executable_hooks:15:in `<main>'
ActiveRecord::StatementInvalid: Mysql2::Error: Can't DROP 'string'; check that column/key exists: ALTER TABLE `lesson_contents` DROP `string
```

For some reason, it's trying to run:
```sql
 ALTER TABLE `lesson_contents` DROP `string`
```

**Comment by iwz** [comment]
Here's the output from a debugging session:

```
From: /Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb @ line 44 Paperclip::Schema::Statements#remove_attachment:

    35: def remove_attachment(table_name, *attachment_names)
    36:   raise ArgumentError, "Please specify attachment name in your remove_attachment call in your migration." if attachment_names.empty?
    37:
    38:   options = attachment_names.extract_options!
    39:
    40:   attachment_names.each do |attachment_name|
    41:     binding.pry
    42:     COLUMNS.each_pair do |column_name, column_type|
    43:       column_options = options.merge(options[column_name.to_sym] || {})
 => 44:       remove_column(table_name, "#{attachment_name}_#{column_name}", column_type, column_options)
    45:     end
    46:   end
    47: end

[1] pry(#<ActiveRecord::ConnectionAdapters::Mysql2Adapter>)> table_name
=> "lesson_contents"
[2] pry(#<ActiveRecord::ConnectionAdapters::Mysql2Adapter>)> "#{attachment_name}_#{column_name}"
=> "asset_file_name"
[3] pry(#<ActiveRecord::ConnectionAdapters::Mysql2Adapter>)> column_type
=> :string
[4] pry(#<ActiveRecord::ConnectionAdapters::Mysql2Adapter>)> column_options
=> {}
[5] pry(#<ActiveRecord::ConnectionAdapters::Mysql2Adapter>)> continue
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

Mysql2::Error: Can't DROP 'string'; check that column/key exists: ALTER TABLE `lesson_contents` DROP `string`/Users/iwz/.rvm/gems/ruby-2.1.6/gems/rack-mini-profiler-0.1.27/Ruby/lib/patches/sql_patches.rb:45:in `query'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/rack-mini-profiler-0.1.27/Ruby/lib/patches/sql_patches.rb:45:in `query'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `block in execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in `block in log'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in `log'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/mysql2_adapter.rb:213:in `execute'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:279:in `block in remove_column'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:278:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:278:in `remove_column'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:44:in `block (2 levels) in remove_attachment'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:42:in `each_pair'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:42:in `block in remove_attachment'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:40:in `each'
/Users/iwz/.rvm/gems/ruby-2.1.6/gems/paperclip-4.3.0/lib/paperclip/schema.rb:40:in `remove_attachment'
```

**Comment by iwz** [comment]
Ah, it seems that in Rails 3.2, the mysql adapter has a different signature for `remove_column`:

https://github.com/rails/rails/blob/3-2-stable/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L639

```ruby
def remove_column(table_name, *column_names)
```

**Comment by iwz** [comment]
This was fixed by #1947
